modin/core/storage_formats/pandas/utils.py

```diff
@@ -38,40 +40,48 @@ def get_default_chunksize(length, num_splits):
     )
 
 
-def compute_chunksize(df_shape, num_splits, default_block_size=32, axis=None):
+def compute_chunksize(
+    row_count=None, col_count=None, num_splits=None, default_block_size=32
+):
     """
     Compute the number of rows and/or columns to include in each partition.
 
     Parameters
     ----------
-    df_shape : tuple
-        Dataframe shape for which chunksizes are calculated.
-    num_splits : int
-        Number of splits to separate the DataFrame into.
+    row_count : int, optional
+        Rows count.
+    col_count : int, optional
+        Columns count.
+    num_splits : int, optional
+        Number of splits to separate the DataFrame into. `NPartitions` by default.
     default_block_size : int, default: 32
         Minimum number of rows/columns in a single split.
-    axis : int, optional
-        Axis to split across. If not specified - split accros both axes.
 
     Returns
     -------
-    int if axis was specified, tuple of ints otherwise,
-        If axis is 1 or 0, returns an integer number of rows/columns to split the
-        DataFrame. If axis is None, returns a tuple containing both.
+    int
+        - `row_count` and `col_count` are specified: returns tuple of ints otherwise,
+        - `row_count`/`col_count` are specified: returns an integer number of rows/columns to split the
+        DataFrame.
     """
-    if axis == 0 or axis is None:
-        row_chunksize = get_default_chunksize(df_shape[0], num_splits)
+    assert row_count is not None or col_count is not None
```

Comment:
```suggestion
    assert not (row_count is None and col_count is None)
```
I think this looks more intuitive, but it's totally up to you to accept or not this proposal - it's pure cosmetics (your version works slightly faster, but I think the difference is negligible)